### ISSUE TITLE
[V3] Implement new Concept Exercise: generator-expressions

### ISSUE THREAD

**Issue by BethanyG** [exercism/python#2292]
This issue describes how to implement the `generator-expressions` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)
- [Python Track concept exercise implementation guide](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md).
- [Python Track "make a concept exercise utility"](https://github.com/exercism/v3/tree/master/languages/python/bin).

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the syntax and variants of `generator expressions` in Python. 

## Learning objectives

-  Understand how a `generator expression` relates to both  a `list comprehension` and a `generator`
-  Understand where `generator expressions` can and cannot be utilized
-  Create a `generator expression`
-  Use a generator expression as an argument to a function such as `''.join()` or `sum()`
-  Use a generator expression as an element of a comprehension

## Out of scope

- Memory and performance characteristics and optimizations
-  Generator methods such as `throw()` and `close()`
-  Using the `assignment expression` (_walrus operator_) with a `generator expression`

## Concepts

-  `generator expressions`

## Prerequisites

- [for-loops](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/for_loops.md)
- [builtin-types](https://github.com/exercism/v3/tree/master/languages/python/reference/concepts/builtin_types)
- [data-types](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/data_structures.md)
- [generators](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/generators.md)
- [iterables](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iterables.md)
-  [iterators](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iterators.md)
- [iteration](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/iteration.md)
- [list-comprehensions](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/list_comprehension.md)
- [comprehension-syntax](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/comprehension_syntax.md)

## Resources to refer to

- [Generator expressions and list comprehensions (Python official docs)](https://docs.python.org/3/howto/functional.html?highlight=generators#generator-expressions-and-list-comprehensions)
-  [Generators (Python official docs)](https://docs.python.org/3/howto/functional.html#generators)
-  [List Comprehensions (Python official docs)](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)
- [Generator Expressions in Python (Dan Bader)](https://dbader.org/blog/python-generator-expressions)

### Hints

- `Generator Expressions` section of the Python docs tutorial:  [Generator expressions and list comprehensions (Python official docs)](https://docs.python.org/3/howto/functional.html?highlight=generators#generator-expressions-and-list-comprehensions)
- [Generator Expressions in Python (Dan Bader)](https://dbader.org/blog/python-generator-expressions)

### After

-  Additional `Generator-iterator methods`, such as `generator.send()` and `generator.throw()`
- `generator expressions`
-  Asynchronous generator functions
- `generators` used as coroutines

## Representer

No changes required.

## Analyzer

No changes rquired.

## Implementing

- Tests should be written using unittest.TestCase, and the test file named list_comprehensions_test.py.
- [How to Implement a Concept Exercise in Python](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md)
- [make-concept-exercise Utility](https://github.com/exercism/v3/tree/master/languages/python/bin)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
